### PR TITLE
Popup: update zIndex on focus to make Popup be in the front

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 1]
```diff
@@ -517,22 +515,44 @@ testModule('visibility', moduleConfig, () => {
...
         const contentZIndex = parseInt($content1.css('zIndex'), 10);
         const wrapperZIndex = parseInt(overlay1.$wrapper().css('zIndex'), 10);
 
-        overlay2.show();
         const $content2 = overlay2.$content();
         assert.strictEqual(parseInt($content2.css('zIndex'), 10), contentZIndex + 1);
         assert.strictEqual(parseInt(overlay2.$wrapper().css('zIndex'), 10), wrapperZIndex + 1);
     });
 
+    test('overlay should get the biggest z-index after addaing focusing class', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]:
```suggestion
    test('overlay should get the biggest z-index after focus', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 2]
```diff
@@ -517,22 +515,44 @@ testModule('visibility', moduleConfig, () => {
...
+        const overlay1 = $('#overlay_1').dxOverlay({ visible: true }).dxOverlay('instance');
+        const overlay2 = $('#overlay_2').dxOverlay({ visible: true }).dxOverlay('instance');
+        const overlay1InitialZIndex = 1501;
+
+        overlay1._toggleFocusClass(true, overlay1.$content());
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]: let's not use private api. We can just call `focus` method

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 3]
```diff
@@ -517,22 +515,44 @@ testModule('visibility', moduleConfig, () => {
...
-        overlay2.show();
         const $content2 = overlay2.$content();
         assert.strictEqual(parseInt($content2.css('zIndex'), 10), contentZIndex + 1);
         assert.strictEqual(parseInt(overlay2.$wrapper().css('zIndex'), 10), wrapperZIndex + 1);
     });
 
+    test('overlay should get the biggest z-index after addaing focusing class', function(assert) {
+        const overlay1 = $('#overlay_1').dxOverlay({ visible: true }).dxOverlay('instance');
+        const overlay2 = $('#overlay_2').dxOverlay({ visible: true }).dxOverlay('instance');
+        const overlay1InitialZIndex = 1501;
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]: let's not use digits in name. It's very difficult to read, especially when the digit is not at the end. `firstOverlayZIndex`

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 4]
```diff
@@ -517,22 +515,44 @@ testModule('visibility', moduleConfig, () => {
...
+
+        overlay1._toggleFocusClass(true, overlay1.$content());
+
+        const contentZIndex = parseInt(overlay1.$content().css('zIndex'), 10);
+
+        assert.strictEqual(contentZIndex, parseInt(overlay2.$content().css('zIndex'), 10) + 1);
+        assert.strictEqual(contentZIndex, overlay1InitialZIndex + 2);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]: Why do we need exactly 2 asserts? Is there a scenario when one is red, but another is green?

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 5]
```diff
@@ -517,22 +515,44 @@ testModule('visibility', moduleConfig, () => {
...
+    });
+
+    test('overlay should get back initial z-index after removing focusing class', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]:
```suggestion
    test('overlay should get back initial z-index after focusout', function(assert) {
```

File: js/ui/overlay/ui.overlay.js [hunk 6]
```diff
@@ -720,6 +720,21 @@ const Overlay = Widget.inherit({
         }
     },
 
+    _toggleFocusClass(isFocused, $element) {
+        this.callBase(isFocused, $element);
```
Comment (ksercs) [inline js/ui/overlay/ui.overlay.js]: let's get rid of the second parameter in this function and just pass `arguments` to the callbase. it's better for extensibility

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**